components/script/dom/readablestreambyobreader.rs

```diff
@@ -29,26 +33,49 @@ use crate::dom::readablestream::ReadableStream;
 use crate::script_runtime::{CanGc, JSContext as SafeJSContext};
 
 /// <https://streams.spec.whatwg.org/#read-into-request>
-#[derive(JSTraceable, MallocSizeOf)]
+#[derive(Clone, JSTraceable, MallocSizeOf)]
 pub enum ReadIntoRequest {
     /// <https://streams.spec.whatwg.org/#byob-reader-read>
     Read(#[ignore_malloc_size_of = "Rc is hard"] Rc<Promise>),
 }
 
 impl ReadIntoRequest {
     /// <https://streams.spec.whatwg.org/#read-into-request-chunk-steps>
-    pub fn chunk_steps(&self, _chunk: RootedTraceableBox<Heap<JSVal>>) {
-        todo!()
+    pub fn chunk_steps(&self, chunk: RootedTraceableBox<Heap<JSVal>>) {
+        // chunk steps, given chunk
+        // Resolve promise with «[ "value" → chunk, "done" → false ]».
+        match self {
+            ReadIntoRequest::Read(promise) => {
+                promise.resolve_native(&ReadableStreamReadResult {
+                    done: Some(false),
+                    value: chunk,
+                });
+            },
+        }
     }
 
     /// <https://streams.spec.whatwg.org/#read-into-request-close-steps>
-    pub fn close_steps(&self, _chunk: Option<RootedTraceableBox<Heap<JSVal>>>) {
-        todo!()
+    pub fn close_steps(&self, chunk: Option<RootedTraceableBox<Heap<JSVal>>>) {
+        // close steps, given chunk
+        // Resolve promise with «[ "value" → chunk, "done" → true ]».
+        match self {
+            ReadIntoRequest::Read(promise) => match chunk {
+                Some(chunk) => promise.resolve_native(&ReadableStreamReadResult {
+                    done: Some(true),
+                    value: chunk,
+                }),
+                None => promise.resolve_native(&()),
+            },
+        }
     }
 
     /// <https://streams.spec.whatwg.org/#read-into-request-error-steps>
-    pub(crate) fn error_steps(&self, _e: SafeHandleValue) {
```

Comment:
I think the better link, because it links to the actual steps, is https://streams.spec.whatwg.org/#ref-for-read-into-request-error-steps